src/mongocrypt-kek.c

```diff
@@ -250,6 +292,9 @@ _mongocrypt_kek_cleanup (_mongocrypt_kek_t *kek)
       bson_free (kek->provider.gcp.key_name);
       bson_free (kek->provider.gcp.key_version);
       _mongocrypt_endpoint_destroy (kek->provider.gcp.endpoint);
+   } else if (kek->kms_provider == MONGOCRYPT_KMS_PROVIDER_KMIP) {
+      bson_free (kek->provider.kmip.key_id);
```

Comment:
Is it possible to get to this bson_free() without key_id being allocated? (Well, it shouldn't matter as it's ok to call this function on a NULL pointer, so I think you're ok.)

Reply:
Yes. But it is OK to call `bson_free` with `NULL`. The same assumption is made for other optional fields (e.g. `kek->provider.gcp.key_version`).